### PR TITLE
rogcat: init at 0.4.7

### LINKED ISSUE #356794: Package request: rogcat
## Metadata

* homepage URL: https://github.com/flxo/rogcat
* source URL: https://github.com/flxo/rogcat
* license: mit
* platforms: linux, darwin, windows (not sure about unix)

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by genga898**
Closes #356794

Add [rogcat](https://github.com/flxo/rogcat) to nixpkgs, an adb logcat wrapper.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ro/rogcat/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  ];
+
+  versionCheckProgramArg = [ "--version" ];
+
+  meta = {
+    description = "Adb logcat wrapper";
+    homepage = "https://github.com/flxo/rogcat";
+    changelog = "https://github.com/flxo/rogcat/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ genga898 ];
```
Comment (pluiedev) [inline pkgs/by-name/ro/rogcat/package.nix:40]: Platforms?

File: pkgs/by-name/ro/rogcat/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    mainProgram = "rogcat";
+  };
+
```
Comment (pluiedev) [inline pkgs/by-name/ro/rogcat/package.nix:43]:
```suggestion
```

**Review by pluiedev — APPROVED** [review]
Very very very tiny nitpicks

**Review by pluiedev — APPROVED** [review]

File: pkgs/by-name/ro/rogcat/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  nativeBuildInputs = [
+    pkg-config
+    versionCheckHook
+  ];
+
+  buildInputs = [
+    libudev-zero
+  ];
+
+  versionCheckProgramArg = [ "--version" ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ro/rogcat/package.nix:33]:
```suggestion
  doInstallCheck = true;
  
  nativeInstallCheckInputs = [
    versionCheckHook
  ];

  versionCheckProgramArg = [ "--version" ];
```

versionCheckHook does not run currently. See ofborg

File: pkgs/by-name/ro/rogcat/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    repo = "rogcat";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-l2zfVt2vm5GTrYs6/0D3EesxxPWSmjf2tGS545766iA=";
+  };
+
+  cargoHash = "sha256-cDAS8mengFgBsq9nTiVAjt7pJhKjj7/F9x8IS6vP2ck=";
+
+  nativeBuildInputs = [
+    pkg-config
+    versionCheckHook
```
Comment (Aleksanaa) [inline pkgs/by-name/ro/rogcat/package.nix:26]:
```suggestion
```

**Pull request merged**